akka-javasdk/src/main/scala/akka/javasdk/impl/JsonRpc.scala

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.impl
+
+import akka.annotation.InternalApi
+import akka.http.scaladsl.model.ContentTypes
+import akka.http.scaladsl.model.HttpEntity
+import akka.http.scaladsl.model.HttpMethods
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.StatusCodes
+import akka.javasdk.impl.serialization.JsonSerializer
+import akka.runtime.sdk.spi.ComponentOptions
+import akka.runtime.sdk.spi.HttpEndpointConstructionContext
+import akka.runtime.sdk.spi.HttpEndpointDescriptor
+import akka.runtime.sdk.spi.HttpEndpointMethodDescriptor
+import akka.runtime.sdk.spi.MethodOptions
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import com.fasterxml.jackson.annotation.JsonInclude.Include
+import com.fasterxml.jackson.databind.JsonMappingException
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.jdk.CollectionConverters._
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+object JsonRpc {
```

Comment:
Perhaps some additional `private[akka]` on these, or don't we use that in SDK? I know it doesn't matter for Java.

Reply:
Should be private/proper internal yes 👍